nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java

```diff
@@ -283,38 +283,21 @@ public boolean isRootGroup() {
     public void startProcessing() {
         readLock.lock();
         try {
-            for (final ProcessorNode node : processors.values()) {
+            findAllProcessors().stream().filter(SCHEDULABLE_PROCESSORS).forEach(node -> {
```

Comment:
It seems like here, rather than finding all processors and putting them into a set, then converting to a stream and filtering, we should allow a Predicate to be passed into the findAllProcessors() method? Same for input/output ports